### ISSUE TITLE
volume_from breaks dependancy of services

### ISSUE THREAD

**Issue by Edke** [kontena/kontena#2672]
In stack when combining `volume_from` and `depends_on`/`links`, services that are configured with `volume_from` are deployed before services that are defined in `depends_on`.

Example:
```yaml
---
stack: kraken/test
version: 1.0.2
services:
  web:
    image: web:latest
    volumes_from:
      - data-1
    depends_on:
      - data
      - db
  db:
    image: db:latest
  data:
    image: data:latest
```
Correct deployment order:

1. data or db
2. db or data
3. web

Currently deployed in order:

1. data
2. web
3. db

This behavior breaks dependency as service `web` is deployed before service `db`.

**Comment by Edke** [comment]
Also additional info, when `volume_from` defined within service, it is deployed right after referenced service and is not mentioned in cli that service was deployed.

**Comment by SpComb** [comment]
The server [`StackDeployWorker`](https://github.com/kontena/kontena/blob/0db3106a7c058952d27a3cc736eec165872b4256/server/app/jobs/stack_deploy_worker.rb#L24) is responsible for deploying the services in the correct order, but it does not deploy the services that are dependent on other services: `skipping deployment of #{service.to_path} because it will be deployed by dependencies`

Instead, the [`GridServiceSchedulerWorker`](https://github.com/kontena/kontena/blob/master/server/app/jobs/grid_service_scheduler_worker.rb#L78) will create deployments for each dependent service once the `data` service has been deployed. However, it does not take the stack service ordering into account.

Perhaps we should replace the `GridServiceSchedulerWorker#deploy_dependant_services` logic and just let the `StackDeployWorker` take care of the `volumes_from` dependencies using the same sorting logic used for the `depends_on` links... I'm not sure why the two are handled differently, except that `volumes_from` can also be used outside of stacks, but I don't see why the server should necessarily deploy dependent non-stack services in that case? Ping @jakolehm 

Workaround is to use either `scope: instance` volumes or stateful services instead of `volumes_from`.

**Comment by Edke** [comment]
@SpComb reason I dont use volumes is that this stack serves staging environment, therefore ideal for not using volume/stateful services for data / db. When new db image is build, it will be redeployed.

**Comment by SpComb** [comment]
Fixed in #3100
